src/cc/table_desc.h

```diff
@@ -93,10 +81,10 @@ class TableDesc {
         leaf_snprintf(nullptr),
         is_shared(false),
         is_extern(false) {}
-  TableDesc(const std::string &name, FileDesc &&fd, int type, size_t key_size,
+  TableDesc(const std::string &name, int fd, int type, size_t key_size,
```

Comment:
That call site implicitly invoked the FileDesc(int) constructor originally. Explicitly adding FileDesc(fd) won't help the compiler after you restore the `FileDesc &&` version.